### PR TITLE
Fix comments and nullability in examples

### PR DESCRIPTION
**Pull request by RGBCube**
# Description

Fixed incorrect nullability checks in examples, and comments aswell.

## Type of change

Please delete options that are not 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] Ran `dart analyze` or `make analyze` and fixed all issues
- [x] Ran `dart format --set-exit-if-changed -l 160 ./lib` or `make format` and fixed all issues
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my changes haven't lowered code coverage

### CODE REVIEW

File: example/create_add_role.dart [hunk 1]
```diff
@@ -18,15 +18,14 @@ void main() async {
...
-
-      // Make sure that message was sent in guild not im dm, because we cant add roles in dms
-      if (e.message.guild != null) {
+      // Make sure that the message was sent in a guild and not in a dm, because we cant add roles in dms
+      if (e.message.guild == null) {
         return;
       }
 
       // Creating role with RoleBuilder. We have to cast `e.message` to GuildMessage because we want to access guild property
       // and generic dont have that.
```
Comment (Rapougnac) [inline example/create_add_role.dart:27]: While you're at it, you can remove these lines as they don't have any meaning now
```suggestion
      // Creating a role with RoleBuilder with a given color.
```

File: example/create_add_role.dart [hunk 2]
```diff
@@ -18,15 +18,14 @@ void main() async {
...
-      final role = await e.message.guild!.getFromCache()!.createRole(RoleBuilder("testRole")..color = DiscordColor.chartreuse);
+      final role = await e.message.guild.getFromCache()!.createRole(RoleBuilder("testRole")..color = DiscordColor.chartreuse);
 
       // Cast message author to member because webhook can also be message author. And add role to user
```
Comment (Rapougnac) [inline example/create_add_role.dart:30]:
```suggestion
      // Add role to member.
```

**Review by Rapougnac — COMMENTED** [review]
LGTM

**Review by Rapougnac — APPROVED** [review]

File: example/channel.dart [hunk 3]
```diff
@@ -18,13 +18,13 @@ void main() async {
...
-      // Make sure that message was sent in guild not in dm, because we cant add roles in dms
-      if(e.message.guild != null) {
+      // Make sure that the message was sent in a guild and not in a dm, because we can only create channels in guilds
+      if (e.message.guild == null) {
         return;
       }
 
       // Get guild object from message
-      final guild = e.message.guild!.getFromCache()!;
+      final guild = e.message.guild.getFromCache()!;
```
Comment (Rapougnac) [inline example/channel.dart:27]: Please, keep the cast as is, dart does not promote fields
```suggestion
      final guild = e.message.guild!.getFromCache()!;
```

File: example/create_add_role.dart [hunk 4]
```diff
@@ -18,15 +18,14 @@ void main() async {
...
-      if (e.message.guild != null) {
+      // Make sure that the message was sent in a guild and not in a dm, because we cant add roles in dms
+      if (e.message.guild == null) {
         return;
       }
 
       // Creating role with RoleBuilder. We have to cast `e.message` to GuildMessage because we want to access guild property
       // and generic dont have that.
-      final role = await e.message.guild!.getFromCache()!.createRole(RoleBuilder("testRole")..color = DiscordColor.chartreuse);
+      final role = await e.message.guild.getFromCache()!.createRole(RoleBuilder("testRole")..color = DiscordColor.chartreuse);
```
Comment (Rapougnac) [inline example/create_add_role.dart:28]:
```suggestion
      final role = await e.message.guild!.getFromCache()!.createRole(RoleBuilder("testRole")..color = DiscordColor.chartreuse);
```

File: example/kick_ban.dart [hunk 5]
```diff
@@ -29,34 +29,33 @@ void main() {
...
+      if (e.message.guild == null) {
         return;
       }
 
       // Get user to ban
       final userToBan = getUserToKickOrBan(e.message);
 
       // Ban user using variable initialized before
-      await e.message.guild!.getFromCache()!.ban(userToBan);
+      await e.message.guild.getFromCache()!.ban(userToBan);
```
Comment (Rapougnac) [inline example/kick_ban.dart:41]:
```suggestion
      await e.message.guild!.getFromCache()!.ban(userToBan);
```

File: example/kick_ban.dart [hunk 6]
```diff
@@ -29,34 +29,33 @@ void main() {
...
+      if (e.message.guild == null) {
         return;
       }
 
       // Get user to kick
       final userToBan = getUserToKickOrBan(e.message);
 
       // Kick user
-      await e.message.guild!.getFromCache()!.kick(userToBan);
+      await e.message.guild.getFromCache()!.kick(userToBan);
```
Comment (Rapougnac) [inline example/kick_ban.dart:58]:
```suggestion
      await e.message.guild!.getFromCache()!.kick(userToBan);
```

**Review by Rapougnac — DISMISSED** [review]

**Comment by abitofevrything** [comment]
Change looks good, it'd be great if you would fix the other issues brought up by the checks if you have time :)

**Comment by RGBCube** [comment]
Done, though since i did it through the web ui theres lots of commits, squashing would work

**Review by abitofevrything — APPROVED** [review]

File: example/channel.dart (hunk 3 above)
Comment (RGBCube) [reply example/channel.dart]: > Please, keep the cast as is, dart does not promote fields

That is weird, is there a proposal for it?

Comment (Rapougnac) [reply example/channel.dart]: Not a proposal, but a general issue; https://github.com/dart-lang/language/issues/1415

**Pull request merged**